monitoring/src/main/scala/akkaviz/server/WebSocketRepl.scala

```diff
@@ -0,0 +1,178 @@
+        }
+      }
+
+      def withConsoleRedirection(fun: => Any): Unit = {
+        Console.withIn(in) {
+          Console.withOut(sshOut) {
+            Console.withErr(sshOut) {
+              fun
+            }
+          }
+        }
+      }
+    }
+
+    val wsFlow: Flow[Message, Message, _] = {
+
+      val in = Flow[Message].flatMapConcat {
+        case b: BinaryMessage => b.dataStream
+        case t: TextMessage   => t.textStream.map(str => ByteString(str.getBytes))
+      }.toMat(StreamConverters.asInputStream(1.hour))(Keep.right)
+
+      val out = StreamConverters.asOutputStream(1.hour)
+        .groupedWithin(1000, 100.millis)
+        .map(_.reduce(_ ++ _))
+        .map[Message](bs => BinaryMessage.Strict(bs))
+
+      Flow.fromSinkAndSourceMat(in, out)(Keep.both).mapMaterializedValue {
+        case (in, out) => runRepl(this.getClass.getClassLoader, in, out)
+      }.keepAlive(30.seconds, () => BinaryMessage.Strict(ByteString()))
+
+    }
+
+    Directives.handleWebSocketMessages(wsFlow)
+
+  }
+
+  private class SshOutputStream(out: OutputStream) extends OutputStream {
+    override def close() = {
+      out.close()
+    }
+
+    override def flush() = {
+      out.flush()
+    }
+
+    override def write(byte: Int): Unit = {
+      if (byte.toChar == '\n') out.write('\r')
+      out.write(byte)
+    }
+
+    override def write(bytes: Array[Byte]): Unit = for {
+      i ← bytes.indices
+    } write(bytes(i))
+
+    override def write(bytes: Array[Byte], offset: Int, length: Int): Unit = {
+      write(bytes.slice(offset, offset + length))
+    }
+  }
+
+  /**
+   * Container for staging environment important for Ammonite repl to run correctly.
+   *
+   * @param thread             a thread where execution takes place. Important for restoring contextClassLoader
+   * @param contextClassLoader thread's context class loader. Ammonite repl uses that to load classes
+   * @param systemIn
+   * @param systemOut
+   * @param systemErr
+   */
+  private case class Environment(
+    thread: Thread,
+    contextClassLoader: ClassLoader,
+    systemIn: InputStream,
+    systemOut: PrintStream,
+    systemErr: PrintStream
+  )
+
+  private object Environment {
+    def apply(classLoader: ClassLoader, in: InputStream, out: PrintStream): Environment =
+      apply(Thread.currentThread(), classLoader, in, out, out)
+
+    def apply(classLoader: ClassLoader, in: InputStream, out: OutputStream): Environment =
+      apply(classLoader, in, new PrintStream(out))
+
+    /**
+     * Collects information about current environment
+     */
+    def collect() = Environment(
+      Thread.currentThread(),
+      Thread.currentThread().getContextClassLoader,
+      System.in,
+      System.out,
+      System.err
+    )
+
+    /**
+     * Runs your code with supplied environment installed.
+     * After execution of supplied code block will restore original environment
+     */
+    def withEnvironment(env: Environment)(code: ⇒ Any): Any = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/lustefaniak/akka-viz/file/2101331339/issues/source?bid=3121282&fileBranchId=3142893#l155)